tools/src/main/java/it/eng/tools/configuration/ApplicationPropertiesConfiguration.java

```diff
@@ -0,0 +1,91 @@
+package it.eng.tools.configuration;
+
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.env.OriginTrackedMapPropertySource;
+import org.springframework.core.env.AbstractEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.PropertySource;
+import org.springframework.stereotype.Component;
+
+import it.eng.tools.model.ApplicationProperty;
+import it.eng.tools.repository.ApplicationPropertiesRepository;
+import it.eng.tools.service.ApplicationPropertiesService;
+import jakarta.annotation.PostConstruct;
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class ApplicationPropertiesConfiguration {
+
+	@Autowired
+	private Environment environment;
+
+
+	private final ApplicationPropertiesService service;
+
+	private final ApplicationPropertiesRepository repository;
+
+
+	public ApplicationPropertiesConfiguration(ApplicationPropertiesService service, ApplicationPropertiesRepository repository) {
+		super();
+		this.repository = repository;
+		this.service = service;
+	}
+
+	@PostConstruct
+	public void init() {
+		log.info("init() is running");
+		//ApplicationPropertiesService service = new ApplicationPropertiesService(repository);
+
+		PropertySource propertySource = ((AbstractEnvironment) environment)
+				.getPropertySources().stream()
+				.filter(i -> i instanceof OriginTrackedMapPropertySource)
+				.filter(i -> i.getName().contains("application.properties"))
+				.findFirst()
+				.orElseThrow();
+		Map<String, String> properties = (Map<String, String>) propertySource.getSource();
+
+		log.info("Size:" + properties.size());
+
+		//List<ApplicationProperty> propertiesToStore = new ArrayList<ApplicationProperty>(properties.size());
+
+		Set<Entry<String, String>> entrySet = properties.entrySet();
+		entrySet.forEach(entry -> {
+			String key = entry.getKey();
+			Object value = entry.getValue();
+			log.info(entry.getKey() + "=" + value);
+
+			ApplicationProperty applicationProperty = managePropertyOnEnv(key, value);
+
+			log.info(applicationProperty.toString());
+
+		});
+	}
+	
+	private ApplicationProperty managePropertyOnEnv(String key, Object value) {

```

Comment:
Why are you inserting properties in mongo?
I'm guessing all properties from application.properties file.
This should not be done, only those properties that are found in init_data.json file.
Properties in application.properties are "static" properties, and must be in file; updating those properties in file, will require to restart connector (stop, change property and start again)

Reply:
The logic is a little different.
At first startup the ApplicationPropertiesConfiguration reads all the application properties from the env and inserts them into mongo. In this way all the properties of the TC exist in the mongodb collection even if they were not present in the init_data.json. 
If the same property exists both in Mongo and in the env, the value that is considered is the Mongo one.
But to avoid a repeated reading of these values ​​from the db (@davidjovanovic asked for it) the changed values ​​are also reinserted as properties in the environment in a PropertySource  called "storedApplicationProperties" which has a higher priority than application properties.